next/frontend/utils/amplify.ts

```diff
@@ -70,3 +75,43 @@ export const getCurrentAuthenticatedUser = async () => {
     return null
   }
 }
+
+const requiredModules = [
+  // API cannot function without Auth
+  Auth,
+  // Auth cannot function without Credentials
+  Credentials,
+]
+
+// These modules have been tested with SSR
+const defaultModules = [API, Auth, DataStore]
+
+interface Context extends Pick<NextPageContext, 'req'> {
+  modules?: any[]
+}
+
+export function withSSRContext(context: Context = {}) {
```

Comment:
same as the other comment - if modelled after something, please reference it here